src/cunumeric/index/advanced_indexing.cu

```diff
@@ -29,95 +29,86 @@ template <typename Output, typename Pitches, typename Point, int32_t DIM>
 static __global__ void __launch_bounds__(THREADS_PER_BLOCK, MIN_CTAS_PER_SM)
   count_nonzero_kernel(size_t volume,
                        Output out,
+                       Buffer<int64_t> offsets,
                        AccessorRO<bool, DIM> index,
                        Pitches pitches,
                        Point origin,
                        size_t iters,
-                       Buffer<int64_t> offsets)
+                       const size_t skip_size,
+                       const size_t key_dim)
 {
   size_t value = 0;
-  for (size_t idx = 0; idx < iters; idx++) {
-    const size_t offset = (idx * gridDim.x + blockIdx.x) * blockDim.x + threadIdx.x;
-    if (offset < volume) {
-      auto point      = pitches.unflatten(offset, origin);
-      auto val        = static_cast<size_t>(index[point]);
-      offsets[offset] = val;
-      SumReduction<size_t>::fold<true>(value, val);
+  for (size_t i = 0; i < iters; i++) {
+    size_t idx = (i * gridDim.x + blockIdx.x) * blockDim.x + threadIdx.x;
+    if (idx > volume) break;
+    auto point = pitches.unflatten(idx, origin);
+    size_t val;
+    if (index[point] and ((idx + 1) % skip_size == 0)) {
+      offsets[idx] = 1;
```

Comment:
I guess you could rewrite this as follows to avoid control divergence:
```
bool val = index[point] and ((idx + 1) % skip_size == 0);
offsets[idx] = static_cast<int64_t>(val);
```
(a smart compiler would do this conversion for us, but there's no guarantee)

Reply:
Thank you!